### PR TITLE
Automation test for Wiki search results show HTML source when document includes a comma

### PR DESCRIPTION
**Pull request by labkey-sweta**
#### Rationale
Automation test for Wiki search results show HTML source when document includes a comma

Repro:

- Create an HTML wiki page with text "This is my HTML, with commas" on labkey.org (e.g., in the Staff project).
- Save
- Search for "commas" in the current folder
- Search result should not show HTML tags

https://www.labkey.org/home/Developer/issues/Secure/issues-details.view?issueId=47814

#### Related Pull Requests
* 

#### Changes
*

### CODE REVIEW

File: src/org/labkey/test/tests/wiki/WikiTest.java [hunk 1]
```diff
@@ -198,6 +202,32 @@ public void testShowPageTreeForWiki()
         assertElementPresent(Locator.id("wiki-toc-tree").append(Locator.linkContainingText(WIKI_PAGE_TITLE + " (" + WIKI_PAGE_NAME + ")")));
     }
 
+    /*
+        Regression coverage for
+         :https://www.labkey.org/home/Developer/issues/Secure/issues-details.view?issueId=48019
```
Comment (labkey-jeckels) [inline src/org/labkey/test/tests/wiki/WikiTest.java]:
```suggestion
         https://www.labkey.org/home/Developer/issues/Secure/issues-details.view?issueId=48019
```

**Review by labkey-jeckels — APPROVED** [review]

File: src/org/labkey/test/tests/wiki/WikiTest.java [hunk 2]
```diff
@@ -39,7 +42,8 @@
 public class WikiTest extends BaseWebDriverTest
 {
     private static final String PROJECT_NAME = TRICKY_CHARACTERS_FOR_PROJECT_NAMES + "WikiVerifyProject";
-
+    private static final SearchResultsQueue SEARCH_RESULTS_QUEUE = new SearchResultsQueue();
+    private final SearchHelper _searchHelper = new SearchHelper(this, SEARCH_RESULTS_QUEUE).setMaxTries(6);
```
Comment (labkey-tchad) [inline src/org/labkey/test/tests/wiki/WikiTest.java]: These don't need to be class members if only one test is using them.

File: src/org/labkey/test/tests/wiki/WikiTest.java [hunk 3]
```diff
@@ -198,6 +202,32 @@ public void testShowPageTreeForWiki()
...
+        String wikiContent = "This is my HTML, with commas";
+
+        goToProjectHome();
+        log("Creating the wiki " + wikiTitle);
+        WikiHelper wikiHelper = new WikiHelper(this);
+        wikiHelper.createNewWikiPage("HTML");
+        numberOfWikiCreated++;
+        setFormElement(Locator.name("name"), wikiName);
+        setFormElement(Locator.name("title"), wikiTitle);
+        wikiHelper.setWikiBody(wikiContent);
```
Comment (labkey-tchad) [inline src/org/labkey/test/tests/wiki/WikiTest.java]: If I remember correctly, the wiki helper doesn't know how to use the graphical editor. It switches to the html source tab automatically. If so, this needs to have some html tags added explicitly.
```suggestion
        wikiHelper.setWikiBody("<p>" + wikiContent + "</p>");
```

File: src/org/labkey/test/tests/wiki/WikiTest.java [hunk 4]
```diff
@@ -198,6 +202,32 @@ public void testShowPageTreeForWiki()
...
+        wikiHelper.saveWikiPage();
+
+        _searchHelper.searchFor("commas");
```
Comment (labkey-tchad) [inline src/org/labkey/test/tests/wiki/WikiTest.java]: Make sure you're only searching within the project for this test. Don't want other tests interfering with this.

**Review by labkey-tchad — CHANGES_REQUESTED** [review]

File: src/org/labkey/test/tests/wiki/WikiTest.java [hunk 5]
```diff
@@ -198,6 +198,33 @@ public void testShowPageTreeForWiki()
...
+
+        goToProjectHome();
+        log("Creating the wiki " + wikiTitle);
+        WikiHelper wikiHelper = new WikiHelper(this);
+        wikiHelper.createNewWikiPage("HTML");
+        numberOfWikiCreated++;
+        setFormElement(Locator.name("name"), wikiName);
+        setFormElement(Locator.name("title"), wikiTitle);
+        wikiHelper.setWikiBody("<p>" + wikiContent + "</p>");
+        wikiHelper.saveWikiPage();
```
Comment (labkey-tchad) [inline src/org/labkey/test/tests/wiki/WikiTest.java]: Increment this count after saving.
```suggestion
        setFormElement(Locator.name("name"), wikiName);
        setFormElement(Locator.name("title"), wikiTitle);
        wikiHelper.setWikiBody("<p>" + wikiContent + "</p>");
        wikiHelper.saveWikiPage();
        numberOfWikiCreated++;
```

File: src/org/labkey/test/tests/wiki/WikiTest.java [hunk 6]
```diff
@@ -198,6 +198,33 @@ public void testShowPageTreeForWiki()
...
+
+
+        searchFor(PROJECT_NAME, "commas", numberOfWikiCreated, wikiTitle);
```
Comment (labkey-tchad) [inline src/org/labkey/test/tests/wiki/WikiTest.java]: This search should only show one result, right?
```suggestion
        searchFor(PROJECT_NAME, "commas", 1, wikiTitle);
```

**Review by labkey-tchad — CHANGES_REQUESTED** [review]

**Review by labkey-tchad — APPROVED** [review]

**Pull request merged**